Common/SensorFusion/Inc/SensorFusion.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * Sensor Fusion - Converts raw sensor data into human readable formatted structs
+ * Author: Lucy Gong, Dhruv Rawat, Anthony Bertnyk
+ */
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    #include "CControlFunctions.h"
+#ifdef __cplusplus
+}
+#endif
+
+#ifndef SENSORFUSION_HPP
+#define SENSORFUSION_HPP
+
+#include "CommonDataTypes.hpp"
+//Frequency of SF calculations in Hz
+const int SF_FREQ = 200;
+
+// -1 = FAILED
+// 0 = SUCCESS
+// 1 = Old Data
+typedef struct {
+    int errorCode;
+} SFError_t;
+
+typedef struct {
+    float roll, pitch, yaw; //Degrees. Yaw of 180 is north.
+    float rollRate, pitchRate, yawRate; //Degrees/second
+    float airspeed; //m/s
+    float altitude; //m
+    float rateOfClimb; //m/s
+    long double latitude; //Decimal degrees
+    float latitudeSpeed; //m/s
+    long double longitude; //Decimal degrees
+    float longitudeSpeed; //m/s
+    double track; //Degrees. Track of 0 is north.
+    float groundSpeed; //m/s
+    double heading; //Degrees. Heading of 0 is north.
+} SFOutput_t;
+
+//Following structs store the raw sensor data so other modules can have direct access to them without including sensor header files
+typedef struct
+{
+    float magx, magy, magz;
+    float accx, accy, accz;
+    float gyrx, gyry, gyrz; 
+    float temp;
+    
+    bool isDataNew; 
+    enum SensorErrorCodes sensorStatus; 
+    float utcTime; 
+} IMU_Data_t;
+
+typedef struct 
+{
+    double airspeed;        
+
+    enum SensorErrorCodes sensorStatus;       
+    bool isDataNew;         
+    float utcTime;          
+} Airspeed_Data_t;
+
+typedef struct
+{
+    long double latitude;  // 8 Bytes
+    long double longitude; // 8 Bytes
+    float utcTime;     // 4 Bytes. Time in seconds since 00:00 (midnight)
+    float groundSpeed; // in m/s
+    int altitude; // in m
+    int16_t heading; // in degrees. Should be between 0-360 at all times, but using integer just in case
+    uint8_t numSatellites;    // 1 Byte
```

Comment:
how will this work on safety, which doesn't have a GPS? Won't you need header guards/directives?

Reply:
This struct is only used inside the AUTOPILOT guard. Maybe it would be good to guard declarations too, but I also don't want to clutter too much with preprocessor stuff.

Reply:
Makes sense, ty